src/kakarot/precompiles/precompiles.cairo

```diff
@@ -41,9 +54,21 @@ namespace Precompiles {
     }(evm_address: felt, input_len: felt, input: felt*) -> (
         output_len: felt, output: felt*, gas_used: felt, reverted: felt
     ) {
-        // Compute the corresponding offset in the jump table:
-        // count 1 for "next line" and 3 steps per precompile evm_address: call, precompile, ret
-        tempvar offset = 1 + 3 * evm_address;
+        let is_rollup_precompile_ = is_rollup_precompile(evm_address);
+        if (is_rollup_precompile_ != 0) {
+            // Rollup precompiles start at address 0x100. The last ethereum precompile is at address 0x0a.
+            // The offset is computed as relative to the last ethereum precompile.
+            tempvar offset = 1 + 3 * (
+                (LAST_ETHEREUM_PRECOMPILE_ADDRESS + 1) +
+                (evm_address - FIRST_ROLLUP_PRECOMPILE_ADDRESS)
+            );
+            tempvar range_check_ptr = range_check_ptr;
+        } else {
+            // Compute the corresponding offset in the jump table:
+            // count 1 for "next line" and 3 steps per precompile evm_address: call, precompile, ret
+            tempvar offset = 1 + 3 * evm_address;
+            tempvar range_check_ptr = range_check_ptr;
+        }
```

Comment:
why do you need to  `tempvar range_check_ptr` ?

also, I'd compute in the `if/else` only the variable part

```
// the precompiles jump tables is ethereum precompiles then rollup precompiles
// Ethereum precompiles addresses are in range 0, LAST_ETHEREUM_PRECOMPILE_ADDRESS
// Rollup precompiles addresses are in range FIRST_ROLLUP_PRECOMPILE_ADDRESS, LAST_ETHEREUM_PRECOMPILE_ADDRESS

if (is_rollup_precompile_ != 0) {
    tempvar index = evm_address - FIRST_ROLLUP_PRECOMPILE_ADDRESS + LAST_ETHEREUM_PRECOMPILE_ADDRESS + 1;
} else {
    tempvar index = evm_address;
}

// Compute the corresponding offset in the jump table:
// count 1 for "next line" and 3 steps per precompile: call, precompile, ret
tempvar offset = 1 + 3 * index;
```